### DISCUSSION TITLE
Climate - Heat/Cool colors

### DISCUSSION THREAD

**Discussion by Matze89x** [home-assistant/frontend#19499] [Theming]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.
- [ ] I have tried reproducing the issue in [safe mode](https://www.home-assistant.io/blog/2023/11/01/release-202311/#restarting-into-safe-mode) to rule out problems with unsupported custom resources.

### Describe the issue you are experiencing

In a nutshell: Could it be that the colors of Climate are the wrong way round in the view? Low is blue and high is red?

### Describe the behavior you expected

Climate entity with low and high temperature selection

### Steps to reproduce the issue

1.Click on climate entity with low and high temperature selection
2.
3.
...

### What version of Home Assistant Core has the issue?

2024.1.3

### State of relevant entities

```txt
climate with heat and low

climate.vaillant
hvac_modes: off, cool, heat, auto
min_temp: 5
max_temp: 30
target_temp_step: 0.5
current_temperature: 36.5
target_temp_high: 20
target_temp_low: 17
friendly_name: Vaillant Heizung
supported_features: 2
```
![image](https://github.com/home-assistant/frontend/assets/75788739/b071a027-63a9-4d3f-b355-98523dc2ef86)

### Additional information

--high-color: var(--state-climate-cool-color, var(--state-climate-active-color, var(--state-active-color)));
    --low-color: var(--state-climate-heat-color, var(--state-climate-active-color, var(--state-active-color)));
}

why is in css high-color with climate cool?

**Comment by karwosts** [comment]
I think it makes sense when you consider that is the area where the cooling will activate. 

So if you think of the current temperature as a dot on the curve, when that dot enters the blue region, the AC (cooling) will activate, and opposite for when it enters the red ("heating") area.

**Comment by Matze89x** [comment]
Sorry, but I still don't understand, because when I operate the red area, do I actually operate cooling in red and heating in blue? Let's be honest, even if it's air conditioning or a heater - blue is cold and red is hot, right? ;-)

By the way: if I only create individual entities for hot and cold, the colours match.

**Comment by matthiasdebaat** [comment]
In your screenshot I see a current temperature of 36.5, target temperature of 20, mode `heat` and it is idle, because the target has been reached. I'm not sure where the current dot in the slider is. The temperature may be too high to fall outside this display. @piitaya what do you think?

The colors are correct. Red symbolizes heating and blue cooling. That's also what the label in the middle will show. Here is an example of different scenarios.
![CleanShot 2024-01-18 at 09 40 52@2x](https://github.com/home-assistant/frontend/assets/20203284/3f2c8cdf-711d-41c1-b91d-ccbaa802c8f9)

Can you explain a little more about what you expected?

**Comment by piitaya** [comment]
High temperature is the temperature used for cooling and low temperature is the temperature used for heating.
In your example you have 17C° for low temp and 20°C for high temp.
When the temperature is below 17°C (e.g. 15°C), it will heat to 17°C
When the temperature is above 20°C (e.g. 22°C), it will cool to 20°C
When the temperature is between 17°C and 20°C, it will do nothing.
So, in conclusion, the low temperature is used for heating and the high for cooling. That's why we have this color code.

**Comment by Matze89x** [comment]
Hello, thank you very much for the clarification and I understand your point of view.

First of all, what is behind heat and cool and the current temperature. It is a gas heating system where I set the night setback and the day temperature. The current temperature tells me how high the flow temperature to the heating system is.

I'll go into their logic, then I'd like to take the subject a little further. Would it make sense to add an attribute for swapping the color? It would be very desirable.
In my case, the current value would not be coupled with low and high
Night temperature: 18°C - flow goes to approx. 35°C (calculated according to outside temperature)
Day temperature: 20°C - flow goes to approx. 38°C (calculated depending on the outside temperature)

In my case, I would be happy to be able to make the day and night setting here with a climate - so low in blue would also make sense and high in red.
